tests/external/test_notification_handler.py

```diff
@@ -0,0 +1,101 @@
+                     "Subject" : "Amazon S3 Notification",
+                     "Message" : "{\"Records\":[{\"eventVersion\":\"2.0\",\"eventSource\":\"aws:s3\",\"awsRegion\":\"us-east-1\",\"eventTime\":\"2018-07-03T13:07:40.841Z\",\"eventName\":\"ObjectCreated:Put\",\"userIdentity\":{\"principalId\":\"AWS:AIDAI277OR7TSQEWOQ6SU\"},\"requestParameters\":{\"sourceIPAddress\":\"107.15.167.104\"},\"responseElements\":{\"x-amz-request-id\":\"480DC7730E2843BC\",\"x-amz-id-2\":\"FLaf0a25Q+XMAWgZ+PcF/vX0qjTxQLI8NVO1sFUn9FmO9o6RnHaASueXjzYSxk+pS96+3rjr+Rc=\"},\"s3\":{\"s3SchemaVersion\":\"1.0\",\"configurationId\":\"CURUpdateEvent\",\"bucket\":{\"name\":\"bucket-sns-test\",\"ownerIdentity\":{\"principalId\":\"A2V75OLLZ1ABF7\"},\"arn\":\"arn:aws:s3:::bucket-sns-test\"},\"object\":{\"key\":\"/cur/20180701-20180801/58d25daa-bcb7-4d6c-87cb-6d1f719428ef/cur-Manifest.json\",\"size\":4799,\"eTag\":\"dbd34ba7d518a7a28d34fb05d0e4261f\",\"sequencer\":\"005B3B751CC1E30AA9\"}}}]}",
+                     "Timestamp" : "2018-07-03T13:07:40.924Z",
+                     "SignatureVersion" : "1",
+                     "Signature" : "VOHSWykhwa3UJ9o0QpZHQm6VyxYXF2SMypzQW4DT7GveCrByef02+Xb/j+Fjh2D7kAyFgwq30l2HJc1W048GJ26Xju+x5YRt2KciU88Rra/L97HZmT3lUO3CBWv72nc157QgE1pazuHymjvB4ARaw8Ga1QW4brYBKeyjSL3q8RMXpELnapCc0OzWZ7/tn0sYNbEnro7ZVOAo2NdUgBrNedV/Cu6zjJ2o3k+bYvH9sHN9qI8hGteSg2nshBugTg/7873QbeN5LMaUljTEDnIBaIKdxRLdlj3Q2N0KlFHyMsu4TumqBY2xzEoCa5k5hx7J5F1nerZb0BikwGNGOHkDvA==",
+                     "SigningCertURL" : "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-eaea6120e66ea12e88dcd8bcbddca752.pem",
+                     "UnsubscribeURL" : "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:589179999999:CostUsageNotificationDemo:3affeac1-5249-49a0-876d-a163fb53a004"}
+        body = json.dumps(body_dict)
+
+        with self.assertRaises(NotificationHandlerError) as error:
+            _ = NotificationHandler(headers, body)
+        self.assertTrue('Unexpected message type' in str(error.exception))
+
+    def test_initializer_unsupported_provider(self):
+        """Test to initializer unknown provider."""
+        headers = [('X-Wizz-Bang-Message-Type', 'Notification'),
+                   ('Content-Length', 761),
+                   ('Content-Type', 'text/plain; charset=UTF-8'),
+                   ('Connection', 'Keep-Alive'),
+                   ('User-Agent', 'Amazon Simple Notification Service Agent')]
+
+        body_dict = {"Type" : "Notification",
+                     "MessageId" : "869ace9b-691a-5148-a22c-9d2e1e46e3de",
+                     "Subject" : "Wizz Bang Notification",
+                     "Timestamp" : "2018-07-03T13:07:40.924Z"}
+        body = json.dumps(body_dict)
+
+        with self.assertRaises(NotificationHandlerError) as error:
+            _ = NotificationHandler(headers, body)
+        self.assertTrue('Unsupported cloud provider.' in str(error.exception))
```

Comment:
Same as before - if error.exception isn't an Iterable, this should probably be an `assertEqual`